html2docx.py

```diff
@@ -14,138 +15,108 @@ def __init__(self, title: str):
-            self.extra_data = ""
-        elif tag in ["b", "strong"]:
-            self.bold -= 1
-        elif tag in ["div", "h1", "h2", "h3", "h4", "h5", "h6", "li", "p"]:
-            self.finish()
-            self.p = None
-        elif tag in ["em", "i"]:
-            self.italic -= 1
-        elif tag in ["ol", "ul"]:
-            self.list_style.pop()
-        elif tag == "u":
-            self.underline -= 1
+            self._add_list_style("List Bullet")
 
     def handle_data(self, data: str) -> None:
-        if self.p or data.strip():
-            self.out(data)
-
-    def out(self, data: str, strip_whitespace: bool = True) -> None:
-        if not self.p:
-            self.p = self.doc.add_paragraph()
-
-        if self.extra_data:
-            data += self.extra_data
-
-        run = Run(
-            data,
-            bold=bool(self.bold),
-            italic=bool(self.italic),
-            underline=bool(self.underline),
-            strip_whitespace=strip_whitespace,
-        )
-        self.runs.append(run)
-
-    def finish(self) -> None:
-        """Collapse white space across runs."""
-        prev_run = None
-        for run in self.runs:
-            if run.strip_whitespace:
-                run.text = WHITESPACE_RE.sub(" ", run.text.strip())
-                if prev_run and prev_run.strip_whitespace:
-                    if prev_run.needs_space_suffix:
-                        prev_run.text += " "
-                    elif run.text and run.needs_space_prefix:
-                        run.text = " " + run.text
-            prev_run = run
-
-        for run in self.runs:
-            if run.text:
-                doc_run = self.p.add_run(run.text)
-                doc_run.bold = run.bold
-                doc_run.italic = run.italic
-                doc_run.underline = run.underline
-
-        self.p = None
-        self.runs = []
-
-    def add_list_style(self, style: str) -> None:
-        self.finish()
-        if self.list_style:
-            # The template included by python-docx only has 3 list styles.
-            level = min(len(self.list_style) + 1, 3)
-            style = f"{style} {level}"
-        self.list_style.append(style)
-
-
-class Run:
-    def __init__(
-        self,
-        text: str,
-        bold: bool = False,
-        italic: bool = False,
-        underline: bool = False,
-        strip_whitespace: bool = True,
-    ):
-        self.text = text
-        self.bold = bold
-        self.italic = italic
-        self.underline = underline
-        self.strip_whitespace = strip_whitespace
-
-        self.needs_space_prefix = bool(WHITESPACE_RE.match(text[0]))
-        self.needs_space_suffix = bool(WHITESPACE_RE.match(text[-1]))
+        if not data.strip() and self.p is None:  # whitespace between block tags
+            return
+        if not self.pre:
+            data = re.sub(WHITESPACE_RE, " ", data)
+        if self.collapse_space:
+            # https://www.w3.org/TR/css-text-3/#collapse
```

Comment:
It is not clear to me why CSS rules to interpret whitespaces are linked here?

Perhaps explain the rational in the comment and mention the handling of whitespaces was inpired by `whitespace: normal;`, if that is what you mean?

Reply:
My question is, how does CSS whitespace handling relate to the `collapse_space` attribute in `HTML2Docx`?

Reply:
I understand how the variable is used, but I fail to understand how it relates to CSS whitespace handling rules?

Reply:
I am not asking what it does, I am asking why there is a comment linking to CSS whitespace handling rules?